searxng/searxng
# Search on category select without JavaScript

## What does this PR do?

* The implementation is change to work without javascript: When enabled, the HTML page contains buttons instead of checkboxes.
* Since JavaScript is not required anymore for it, the setting is moved to the ui preferences and the plugin for it, that used JavaScript, got removed

## Why is this change important?

SearXNG have two modes to pick the categories:
* one as other search engine: when clicked, it sends the user query
* multiple categories: the user can pick multiple categories and then send the query.

The HTML always use checkboxes and labels --> some Javascript send the query in the first mode when the user clicks on a category.

This creates to UX issues:
- screen reader and text browser read or show the categories as check boxes.
- users who disable Javascript can't use the first mode.

This PR intends to fix that.

## How to test this PR locally?
- make run : check in the multiple screen configuration (RTL, dark mode, mobile, tablet, desktop).
- I've tested it on Firefox Desktop, Chromium Desktop and Firefox on Android alredy

## Related issues

closes #1414
closes #2709

(the description has been taken from #2709 by @dalf and extended with additional info)
